mafContrib/transientAsciiSEDMetric.py

```diff
@@ -0,0 +1,662 @@
+                their observations in the observation array.
+            transient_end_index: np.array, int
+                Array of the indicies for each transient that are the end of
+                their observations in the observation array.
+        """
+        # Update the maximum possible transients that could have been
+        # observed during survey_duration.
+        self.max_num_transients += np.ceil(
+            self.survey_duration / (self.transient_duration / 365.25)
+        )
+        # Calculate the observation epoch for each transient lightcurve.
+        self.observation_epoch = (
+            dataSlice[self.mjdCol] - self.survey_start + time_shift
+        ) % self.transient_duration
+        # Identify the observations which belong to each distinct transient.
+        self.transient_id = (
+            np.floor(
+                (dataSlice[self.mjdCol] - self.survey_start) / self.transient_duration
+            )
+            + self.transient_id_start
+        )
+        # Set the starting id number for the next phase shift
+        self.transient_id_start = self.transient_id.max()
+        # Find the set of uniquely observed transients
+        unique_transient_id = np.unique(self.transient_id)
+        # Find the starting index for each transient_id.
+        self.transient_start_index = np.searchsorted(
+            self.transient_id, unique_transient_id, side="left"
+        )
+        # Find the ending index of each transient_id.
+        self.transient_end_index = np.searchsorted(
+            self.transient_id, unique_transient_id, side="right"
+        )
+
+    def setup_run_metric_variables(self, dataSlice):
+        """
+        Wrapper function to handle basic initialization of variables used to run
+        this metric.
+
+        Parameters:
+        -----------
+            dataSlice : numpy.array
+                Numpy structured array containing the data related to the visits
+                provided by the slicer.
+
+        Returns (to self):
+        ------------------
+            dataSlice : numpy.array
+                Now sorted in time.
+            survey_duration: float
+                Defaults to the maximum between the chosen slicer and the user
+                specified duration given to the metric.
+            survey_start: float
+                Defaults to user specified, or metric default, however if it is
+                not defined sets to the earliest time in the given slicer.
+        """
+        # Sort the entire dataSlice in order of time.
+        dataSlice.sort(order=self.mjdCol)
+
+        # Check that survey_duration is not larger than the time of
+        # observations we obtained. If it is, then the max_num_transients will
+        # not be accurate.
+        slicer_time_span = (
+            dataSlice[self.mjdCol].max() - dataSlice[self.mjdCol].min()
+        ) / 365.25
+        # Take the maximum time delta, either specified or from the slicer, to
```

Comment:
Potentially a nitpick question - we have a specific class in MAF called a "Slicer" (healpixslicer, etc.) -- is this what you're referring to here with "slicer"? Are you using data from the slicer itself or from the sliver of opsim data (the dataSlice) provided by the slicer?   (looking at the code, I think it's the latter)

If it's data from the slice, not times associated with the slicer itself, would you mind modifying the word used here a bit to avoid confusion? (may I suggest just using 'slice_time_span')?